File: gitdot-api-ts/src/endpoint/oauth/poll-token.ts
import { z } from "zod";
import { TokenResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const PollTokenRequest = z.object({
  device_code: z.string(),
  client_id: z.string(),
});
export type PollTokenRequest = z.infer<typeof PollTokenRequest>;

export const PollTokenResponse = TokenResource;
export type PollTokenResponse = z.infer<typeof PollTokenResponse>;

export const PollToken = {
  path: "/oauth/token",
  method: "POST",
  request: PollTokenRequest,
  response: PollTokenResponse,
} satisfies Endpoint;
export type PollToken = typeof PollToken;

File: gitdot-api/src/endpoint/oauth/poll_token.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::oauth::TokenResource};

pub struct PollToken;

impl Endpoint for PollToken {
    const PATH: &'static str = "/oauth/token";
    const METHOD: http::Method = http::Method::POST;

    type Request = PollTokenRequest;
    type Response = PollTokenResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct PollTokenRequest {
    pub device_code: String,
    pub client_id: String,
}

pub type PollTokenResponse = TokenResource;
